hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/DatanodeUtil.java

```diff
@@ -127,6 +129,31 @@ public static File idToBlockDir(File root, long blockId) {
     return new File(root, path);
   }
 
+  /**
+   * Take an example. We hava a block with blockid mapping to:
+   * "/data1/hadoop/hdfs/datanode/current/BP-xxxx/current/finalized/subdir0/subdir0"
+   * We return "subdir0/subdir0"
+   * @param blockId blockId
+   * @return The two-level subdir name
+   */
+  public static String idToBlockDirSuffixName(long blockId) {
```

Comment:
What's different between `idToBlockDir` and `idToBlockDirSuffixName`?

Reply:
idToBlockDirSuffixName method is used to generate SubDir lock name, just like BlockPool id or  storageuuid.